File: OnlineGroceryOrderingSystemAuthenticationJava/src/main/java/com/alpha/ogo/security/MyUserDetails.java
package com.alpha.ogo.security;

import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

import com.alpha.ogo.domain.User;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;


public class MyUserDetails implements UserDetails {

	private static final long serialVersionUID = 8889981271356041140L;
	private Long userId;
    private String userName; //mobile number
    private String password;
    private boolean active;
    private List<GrantedAuthority> authorities;

    private String email;
    private String fullName;

    public MyUserDetails(User user) {
    	this.userId = user.getId();
        this.userName = user.getPhoneNo();
        this.password = user.getPassword();
        this.active = user.isActive();
        this.authorities = Arrays.stream(user.getRoles().split(","))
                    .map(SimpleGrantedAuthority::new)
                    .collect(Collectors.toList());
        this.email = user.getEmail();
        this.fullName = user.getFullName();
    }

    public Long getUserId() {
		return userId;
	}

	@Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return authorities;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return userName;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return active;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }
}

File: OnlineGroceryOrderingSystemAuthentication/src/main/kotlin/com/allion/ogoss/security/MyUserDetails.kt
package com.allion.ogoss.security

import com.allion.ogoss.domain.User
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.util.*
import java.util.stream.Collectors

//import java.util.stream.Collectors


class MyUserDetails(user: User) :UserDetails {

    var userId: Long? = null
    var userName = ""
    private var pass :String = ""
    var active: Boolean  = false
    var authorities: MutableList<GrantedAuthority>? = null
    var fullName :String = ""
    var email : String = ""

    init {
        this.userId = user.userId
        this.fullName = user.fullName.toString()
        this.email = user.email.toString()
        this.userName= user.phoneNo.toString()
        this.pass = user.password.toString()
        this.active = user.active
        val userArr  =user.roles?.split(",")?.toTypedArray()
        this.authorities = Arrays.stream(userArr).map(::SimpleGrantedAuthority)
            .collect(Collectors.toList())
    }

    override fun getAuthorities(): MutableCollection<out GrantedAuthority> {
        return this.authorities !!
    }


    override fun getPassword(): String {
        return this.pass
    }

    override fun getUsername(): String {
        return this.userName
    }

    override fun isAccountNonExpired(): Boolean {return true}

    override fun isAccountNonLocked(): Boolean {return true}

    override fun isCredentialsNonExpired(): Boolean {return true}

    override fun isEnabled(): Boolean =this.active

}